client/src/featureform/serving.py

```diff
@@ -126,22 +137,82 @@ def _create_channel(self, host, insecure, cert_path):
             return secure_channel(host, cert_path)
 
     def training_set(self, name, variation, include_label_timestamp):
-        return Dataset(self._stub).from_stub(name, variation)
+        return Dataset(self._stub, resource_client=self._resource_client).from_stub(name, variation)
 
     def features(self, features, entities):
+        return FeatureServer(self._stub, self._resource_client, features, entities)
+
+
+class FeatureServer:
+    def __init__(self, stub, resource_client, features, entities):
+        self._stub = stub
+        self._resource_client = resource_client
+        self._features = features
+        self._entities = entities
+        self._feature_list = []
+        if self._all_ready():
+            self._get_features()
+    
+    def __getitem__(self, item):
+        self._feature_list[item]
+
+    def _get_features(self):
         req = serving_pb2.FeatureServeRequest()
-        for name, value in entities.items():
+        for name, value in self._entities.items():
             entity_proto = req.entities.add()
             entity_proto.name = name
             entity_proto.value = value
-        for (name, variation) in features:
+        for (name, variation) in self._features:
             feature_id = req.features.add()
             feature_id.name = name
             feature_id.version = variation
         resp = self._stub.FeatureServe(req)
-        return [parse_proto_value(val) for val in resp.values]
+        self._feature_list = [parse_proto_value(val) for val in resp.values]
+    
+    """Waits for all features to be ready before retrieving them.
```

Comment:
Can you move this after the function definition?

```
def wait(self, timeout=99999):
      """
      Waits for all features to be ready before retrieving them.
      ....
      """

```

Reply:
moved